templates/assistants/assistant_form.html

```diff
@@ -13,9 +13,30 @@
   </div>
 {% endblock %}
 {% block pre_form %}
+  {% if files_out_of_sync %}
+    <div id="sync-notification" class="flex w-full shadow-md !rounded-none alert m-1">
+      <p><strong>Warning:</strong> Your assistant has files that are uploaded to OpenAI but not present in Open Chat Studio. This may lead to unexpected behavior in the assistant. To resolve this issue, please upload the necessary files on this page.</p>
+    </div>
+  {% endif %}
   {% if object.id %}
     <h2 class="text-l pg-text-muted">Assistant ID: {{ object.assistant_id }}</h2>
   {% endif %}
+  <div id="sync-wrapper" class="flex w-full gap-2">
+    <div id="sync-status"
+         hx-get="{% url 'assistants:check_sync_status' request.team.slug object.pk %}"
+         hx-trigger="load"
+         hx-target="#sync-container"
+         class="flex w-full gap-2">
+      <span class="indicator" id="sync-indicator"></span>
+      <span id="sync-message">Checking sync status...</span>
+    </div>
+
+    <div id="sync-container" hx-on="htmx:afterSwap: this.previousElementSibling.remove();">
```

Comment:
Is the call to `previousElementSibling` to remove the `sync-status` div above? I think if you change the `hx-target` to `sync-wrapper` you won't need to do this (maybe also set `hx-swap="innerHTML"`

Reply:
thanks I was playing around with this and that solution is just what I was looking for https://github.com/dimagi/open-chat-studio/pull/774/commits/a93fd4d5b3916ef6ca8632eedeb0217785b86797